mdn/content
# [PWA] PWA reorg

Part of https://github.com/mdn/mdn/issues/280. @estelle , @captainbrosset .

This needs to land alongside https://github.com/mdn/yari/pull/8238, which defines the new sidebar for the PWA docs.

This PR updates the content of https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps to fit into the new organization we've defined. Since we haven't finished updating the content, it includes lots of the existing content and tries to fit it into the new structure, while also exposing the new content in the nav for the first time.

Once this lands we will be able to incrementally add more content into this structure and remove any existing content that we have superseded, until we are done.

So the goal of this PR is to have a place where we can expose the new stuff _and_ the old stuff, as a temporary measure.

To do this I started by looking at the content of the existing sidebar:

https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installable_PWAs
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Re-engageable_Notifications_Push
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Introduction
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Offline_Service_workers
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Loading
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/App_structure
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Add_to_home_screen 
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installing
https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Structural_overview

So of these 9 pages:
- six of them are more or less following the development of the js13kgames sample app, like a tutorial. They're not exactly a tutorial, because the pages mix up guide and tutorial content. But they do walk through js13kgames, and they have next/previous links, like a multipart tutorial So for now I've called this a tutorial. I expect eventually these pages will all be superseded by the new tutorial. This covers the following pages:
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Introduction
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/App_structure
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installable_PWAs
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Re-engageable_Notifications_Push
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Loading
    - https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Offline_Service_workers
- one, https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Add_to_home_screen is I think already superseded by https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable, so I deleted it and redirected there
- one, https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installing, is currently being rewritten by Estelle, so I have left it where it is, so as not to cause merge conflicts. When Estelle's PR is merged we can move it into Guides.
- one, https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Structural_overview, I have moved into Guides, but I think we will eventually rewrite or delete it.

I've also written little landing pages for things like https://pr25888.content.dev.mdn.mozit.cloud/en-US/docs/Web/Progressive_web_apps/Guides.

I have not rewritten the landing page yet, because this PR was already quite complicated.

******

To test this (and the sidebar) you'd need to:

- check out this branch
- check out https://github.com/mdn/yari/pull/8238
- point that yari branch at this content one, using `CONTENT_ROOT`: https://github.com/mdn/yari/blob/main/docs/REVIEWING.md
- run `yarn dev`
- visit http://localhost:3000/en-US/docs/Web/Progressive_web_apps, and the pages linked in the sidebar.
